Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/tangential_relaxation.h

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2015-2021 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Jane Tournois
+
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_TANGENTIAL_RELAXATION_H
+#define CGAL_POLYGON_MESH_PROCESSING_TANGENTIAL_RELAXATION_H
+
+#include <CGAL/license/Polygon_mesh_processing/meshing_hole_filling.h>
+
+#include <CGAL/Polygon_mesh_processing/compute_normal.h>
+#include <CGAL/property_map.h>
+
+#include <CGAL/boost/graph/Named_function_parameters.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+
+#include <tuple>
+#include <type_traits>
+#include <unordered_map>
+
+#ifdef DOXYGEN_RUNNING
+#define CGAL_PMP_NP_TEMPLATE_PARAMETERS NamedParameters
+#define CGAL_PMP_NP_CLASS NamedParameters
+#endif
+
+
+namespace CGAL {
+namespace Polygon_mesh_processing {
+
+/*!
+* \ingroup PMP_meshing_grp
+* applies an iterative area-based tangential smoothing to the given range of vertices.
+* Each vertex `v` is relocated to its gravity-weighted centroid, and the relocation vector
+* is projected back to the tangent plane to the surface at `v`, iteratively.
+* The connectivity remains unchanged.
+*
+* @tparam TriangleMesh model of `MutableFaceGraph`.
```

Comment:
Actually the graph does not need to be mutable, only the vpm being writable is sufficient.

Reply:
Interesting observation. But I am not sure that we systematically applied that in other functions where we do not change the connectivity.